### PR TITLE
Added Info attributes to ascii class files

### LINKED ISSUE #509: Group common data in Siles into a property
**Describe the feature**

In #500 and in the tbtncSileTBtrans we added the properties `no` and `na` etc. I think these should be removed since they are not sile specific but should be queried in some other way.

Some ideas:
- some property that returns a dictionary or namedtuple
    - this property could be named `info`
    - `settings` (so other things could also be added, like options etc.)
    - ?

The main reason is that the complexity and handling of many properties that are more geometry specific, or setting specific seems to bloat the classes.

Comments?

### PR DESCRIPTION
**Pull request by zerothi**
Now all ascii files will contain the info attributes.

This enables the class to dynamically add attributes based on the content of Sile._info_attributes_.

Each of those entries will be converted to a regex Pattern and will process on each read line by patching the instance readline function.

This is very crude, and might be slow for very big files with lots of attributes, perhaps it can be streamlined a bit by removing those attributes which are static and already found (or perhaps all non-updatable attributes will be removed for searching once the full file has been processed).

It proves quite simple to use and makes the code look prettier and more readable. But the internals are a bit messy.

 - [x] Closes #509
 - [x] Tests added
 - [x] Ranned `isort .` at top--level
 - [ ] Documentation for functionality, `docs/`
 - [x] Changes documented, `CHANGELOG.md`

@tfrederiksen might you have a look, I know this is a drastically different approach than yours, but I think this could be sustainable for other things.

Note that `value` can be anything, meaning that the attribute could be a function that does one thing or the other...

### CODE REVIEW

File: src/sisl/io/sile.py [hunk 1]
```diff
@@ -716,7 +718,188 @@ def close(self):
...
+    _info_attributes_ = []
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.info = _Info(self)
+
+    class _Info:
+        """ The actual .info object that will attached to the instance.
+
+        As of know this is problematic to document.
```
Comment (tfrederiksen) [inline src/sisl/io/sile.py:737]: typo -> "now"

File: src/sisl/io/orca/stdout.py [hunk 2]
```diff
@@ -15,80 +15,39 @@
...
-            self._vdw = True
-        return line
-
-    readline.__doc__ = SileORCA.readline.__doc__
+    _info_attributes_ = [
+        _A("na", r".*Number of atoms",
+           lambda attr, match: int(match.string.split()[-1])),
+        _A("no", r".*Number of basis functions",
+           lambda attr, match: int(match.string.split()[-1])),
+        _A("_vdw_", r".*DFT DISPERSION CORRECTION",
```
Comment (tfrederiksen) [inline src/sisl/io/orca/stdout.py:30]: Why underscore here instead of just `info.vdw`?

**Review by tfrederiksen — CHANGES_REQUESTED** [review]
Overall this looks very good to me; more elegant (but also more complex) than my attempt in #591. I have just some minor comments:
- could you add some critical tests of the new functionality?
- the code still references `self.attribute` instead of `self.info.attribute` in several places, for instance: 
   - https://github.com/zerothi/sisl/blob/d76ae4255c0d4bd773395abd7d4b0ea93c309008/src/sisl/io/orca/stdout.py#L123-L124
   - https://github.com/zerothi/sisl/blob/d76ae4255c0d4bd773395abd7d4b0ea93c309008/src/sisl/io/orca/tests/test_stdout.py#L20-L21
- What about adding the `_info_attributes_` with key quantities also for `siesta`, `vasp` etc?

File: src/sisl/io/orca/stdout.py (hunk 2 above)
Comment (zerothi) [reply src/sisl/io/orca/stdout.py]: Done, changed to `vdw_correction` it was based on the old implementation, but I have now changed to a more meaningful nomenclature. It is not `vdw`, but a `vdw_correction`. Hope you agree.

**Pull request merged**